### ISSUE TITLE
JavaScript custom rule violation is not caught in real project scan

### ISSUE THREAD

**Issue by alexandru-stancioiu** [SonarSource/SonarJS#964]
**I want to report a bug.**

**SonarJS version:** 3.0.0.4962 (also tried the latest version 4.1.0.6085) <br>
**SonarQube version:** 6.7.1 (also tried 7.0)<br>
**Scanner version:** 3.1.0.1141-macosx <br>

**Reproducer** 
I have written a custom rule to catch console.log calls, code below:
```
public class ConsoleLogCheck extends DoubleDispatchVisitorCheck {

  protected static final String RULE_KEY = "CQBP-44---ConsoleLogInJS";
  protected static final String RULE_MESSAGE = "Do not use console.log() for logging";

  @Override
  public void visitCallExpression(CallExpressionTree tree) {
    if (tree.callee().is(org.sonar.plugins.javascript.api.tree.Tree.Kind.DOT_MEMBER_EXPRESSION)) {
      DotMemberExpressionTree callee = (DotMemberExpressionTree) tree.callee();

      if (isCalleeConsoleLogging(callee)) {
        addIssue(callee, RULE_MESSAGE);
      }
    }

    super.visitCallExpression(tree);
  }

  private static boolean isCalleeConsoleLogging(DotMemberExpressionTree callee) {
    return callee.object().is(IDENTIFIER_REFERENCE)
      && "console".equals(((IdentifierTreeImpl) callee.object()).name())
      && "log".equals(callee.property().name());
  }
```
I've also written a unit test that in which the line flagged with `Noncompliant` is caught.
Then, i included the rule in a quality profile and ran a scan (also forced js language on that maven module).
The problem is that even though that file is scanned, SonarQube does not catch the issue.
I did not encounter any such trouble with the Java analyser counterpart.
The sonar-project.properties content is the following:
```
sonar.projectKey=pathlight
sonar.projectName=Pathlight

sonar.modules=core,ui.apps
sonar.java.binaries=target
sonar.javascript.file.suffixes=.js

core.sonar.language=java
core.sonar.projectBaseDir=core
core.sonar.sources=src/main/java

ui.apps.sonar.language=js
ui.apps.sonar.projectBaseDir=ui.apps
ui.apps.sonar.sources=src/main/js
```

**Logs**
```
INFO: -------------  Scan ui.apps
INFO: Base dir: /Users/stancioi/workspace/pathlight/pathlight/ui.apps
INFO: Working dir: /Users/stancioi/workspace/pathlight/pathlight/.scannerwork/pathlight_ui.apps
INFO: Source paths: src/main/js
INFO: Source encoding: UTF-8, default locale: en_US
INFO: Language is forced to js
INFO: Index files
INFO: 1 file indexed
INFO: Quality profile for js: CQ-Rules-JS-Profile
INFO: Sensor SonarJS [javascript]
INFO: 1 source files to be analyzed
INFO: Sensor SonarJS [javascript] (done) | time=96ms
INFO: 1/1 source files have been analyzed
INFO: Sensor SonarJS Coverage [javascript]
INFO: Sensor SonarJS Coverage [javascript] (done) | time=2ms
INFO: Sensor SonarJavaXmlFileSensor [java]
INFO: Sensor SonarJavaXmlFileSensor [java] (done) | time=0ms
```

**Expected behavior**
The rule violation should appear in UI.

**Comment by vilchik-elena** [comment]
Can you provide zip with your custom plugin?
P.S. Do you know there is a rule in SonarJS javascript:S2228 [`Console logging should not be used`](https://rules.sonarsource.com/javascript/RSPEC-2228)

**Comment by alexandru-stancioiu** [comment]
Sure,

[plugin_sonarqube_js_rule.zip](https://github.com/SonarSource/SonarJS/files/1989007/plugin_sonarqube_js_rule.zip)

### How to run
`mvn clean install`
`docker build -t sonarqube-dev .`
`docker run  --name sonarqube-cqrules -p 9002:9000 -p 9092:9092  sonarqube-dev`
SonarQube will be listening on port 9002 for localhost.

I know of the existence of that rule but we wish to write more JS rules in the future and this simple rule is a good exercise.

**Comment by alexandru-stancioiu** [comment]
We figured out that plugin APIs (rule loading for example) for JS are very different than those for Java and we should building a JS plugin from scratch.
